### PR TITLE
[AMS-380] Separate advertisement element "text" type into "plain" and "formatted"

### PR DESCRIPTION
**Pull request by xakep139**

### CODE REVIEW

File: src/MigrationTool/ConverterService.cs [hunk 1]
```diff
@@ -99,28 +99,33 @@ public IElementConstraints GetElementTemplateConstraints(ElementDescriptorType e
...
                         };
-                    }
-                    else
+
+                case ElementDescriptorType.FormattedText:
+                    if (elementTemplate.MaxSymbolsInWord > elementTemplate.TextLengthRestriction)
                     {
-                        return new PlainTextElementConstraints
+                        _logger.LogWarning(
+                            "Element template with {id} has MaxSymbolsInWord larger than TextLengthRestriction ({maxSymbols} and {lengthRestriction}), take the least one",
```
Comment (denisivan0v) [inline src/MigrationTool/ConverterService.cs]: Element template with {id} has MaxSymbolsInWord larger than TextLengthRestriction ({maxSymbols} and {lengthRestriction}), **taking** the least one

File: src/VStore/Objects/ObjectsManagementService.cs [hunk 2]
```diff
@@ -217,7 +217,7 @@ private static void CheckRequredProperties(long id, IObjectPersistenceDescriptor
                                    {
                                        var errors = new List<ObjectElementValidationError>();
                                        var constraints = x.Constraints.For(language);
-                                       var rules = GetVerificationRules(x, constraints);
+                                       var rules = GetVerificationRules(x);
```
Comment (denisivan0v) [inline src/VStore/Objects/ObjectsManagementService.cs]: `GetVerificationRules` -> `GetValidationRules`

**Comment by denisivan0v** [comment]
And one more thing. In `FormattedTextValidator.CheckValidHtml` and similar methods we don't use `IElementConstraints elementConstraints` param. But if, for example, we pass `FormattedTextElementConstraints` object with `ValidHtml = false`, it should be checked.

File: src/MigrationTool/ConverterService.cs (hunk 1 above)
Comment (xakep139) [inline src/MigrationTool/ConverterService.cs]: Done in https://github.com/2gis/nuclear-vstore/pull/43/commits/5452e9f9335bcfa017b8de82096421d766042292

File: src/VStore/Objects/ObjectsManagementService.cs (hunk 2 above)
Comment (xakep139) [inline src/VStore/Objects/ObjectsManagementService.cs]: Done in https://github.com/2gis/nuclear-vstore/pull/43/commits/5452e9f9335bcfa017b8de82096421d766042292

**Comment by xakep139** [comment]
@denisivan0v, now we check all explicit constraints before validating content (https://github.com/2gis/nuclear-vstore/pull/43/commits/5452e9f9335bcfa017b8de82096421d766042292)

**Pull request merged**